sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -2428,17 +2445,23 @@ ESIMD_INLINE SYCL_ESIMD_FUNCTION __ESIMD_NS::simd<T, N> lsc_load_2d(
   constexpr int DstBlockElements = GRFColSize * GRFRowSize;
   constexpr int DstElements = DstBlockElements * NBlocks;
 
+  constexpr uint32_t DstLength = DstBlockElements * sizeof(T) / 32;
```

Comment:
@vmustya - can you please take a look at this block of code and the initialization of 'desc' at L2463 ?

Reply:
According to [bspec](https://gfxspecs.intel.com/Predator/Home/Index/53680), dest size is encoded in units of registers.
```C++
constexpr GrfBytes = 64; /// for PVC&Xe2+ 
constexpr auto DstBlockElements = GrfColSize * GrfRowSize;
constexpr auto DstBlockSize = align(DstBlockElements * sizeof(T), GrfBytes); /// each block is register-aligned, there may be cross-block padding present.

constexpr auto DstElements = std::min(31, DstBlockSize / GrfBytes); /// Dst length of 32 is also encoded as 31.
```